client/thin-replica-client/src/thin_replica_client.cpp

```diff
@@ -304,13 +309,20 @@ void ThinReplicaClient::findBlockHashAgreement(std::vector<bool>& servers_tried,
       }
     }
 
-    readUpdateHashFromStream(server_index, agreeing_subset_members, most_agreeing, most_agreed_block);
+    bool has_hash = readUpdateHashFromStream(server_index, agreeing_subset_members, most_agreeing, most_agreed_block);
+    if (!has_hash) unsuccessful_hash_stream_subset_size++;
     servers_tried[server_index] = true;
 
     if (most_agreeing >= (config_->max_faulty + 1)) {
-      return;
+      return true;
     }
   }
+  auto all_servers_tried =
+      std::all_of(servers_tried.begin(), servers_tried.end(), [](bool elem) { return elem == true; });
+  if (all_servers_tried && unsuccessful_hash_stream_subset_size > servers_tried.size()) {
```

Comment:
Shouldn't it be `>= servers_tried.size()`? E.g., we tried 7 servers and 7 servers returned no hash update. Also, can the amount of failed hashes even be greater than the amount of servers if we only query each server once?

Reply:
Yes, thanks, updated.